decentraland/screens/MyAssetsScreen.js

```diff
@@ -1,42 +1,76 @@
 import React from "react";
 import { connect } from "react-redux";
-import { removeFromMyAssetsList, appendToMyAssetsList } from "../redux/actions";
+import {
+  removeFromMyAssetsList,
+  appendToMyAssetsList,
+  addUserAction,
+} from "../redux/actions";
 import PropTypes from "prop-types";
 import MyAssetsList from "@presentational/MyAssetsList";
 import MyAssetsListItem from "@presentational/MyAssetsListItem";
 import { listsAreEqual, getContracts, logWarn } from "@helpers";
 import { generateAssetFromId } from "@helpers/decentraland";
 import DecentralandUtils from "tasit-sdk/dist/helpers/DecentralandUtils";
-import MyAssetStatus from "@constants/MyAssetStatus";
-const { BOUGHT } = MyAssetStatus;
+import { SUCCESSFUL } from "@constants/UserActionStatus";
 
 export class MyAssetsScreen extends React.Component {
   componentDidMount = async () => {
     const {
       account,
-      myAssets: assetsFromState,
       removeFromMyAssetsList,
       appendToMyAssetsList,
+      addUserAction,
     } = this.props;
     if (account) {
       const { address } = account;
 
-      // Note: If a stored asset doesn't have a status, assuming 'bought'.
-      const boughtAssets = assetsFromState
-        .map(asset => (!asset.status ? { ...asset, status: BOUGHT } : asset))
-        .filter(asset => asset.status === BOUGHT);
+      const boughtAssets = this._getBoughtAssetsFromState();
+
+      const {
+        assetsFromBlockchain,
+        actionsFromBlockchain,
+      } = await this.__getAssetsAndActionsFromBlockchain(address);
 
-      const assetsFromBlockchain = await this._getAssetsFromBlockchain(address);
       const shouldUpdate = !listsAreEqual(assetsFromBlockchain, boughtAssets);
       if (shouldUpdate) {
         // TODO: Add some UI indication that something unexpected happened
         this._logAssetsInconsistency(assetsFromBlockchain, boughtAssets);
+
         removeFromMyAssetsList(boughtAssets);
         appendToMyAssetsList(assetsFromBlockchain);
+        addUserAction(actionsFromBlockchain);
       }
     }
   };
 
+  _getAssetsAndActionsFromBlockchain = async address => {
+    const assetsFromBlockchain = await this._getAssetsFromBlockchain(address);
+
+    const actionsFromBlockchain = assetsFromBlockchain.map(asset => {
+      const { actionId, id: assetId } = asset;
```

Comment:
It's interesting that the `_getAssetsFromBlockchain` function has `actionId` available. I wouldn't have guessed that. I'll need to take another look at that function.

Reply:
(Not gating merging this PR)

Reply:
It comes from `DecentralandUtils`, each `sell order` has its `transactionHash` prop that is being renamed to `actionId` inside of `_getAssetsFromBlockchain` function.

Reply:
Ah right, I think that makes sense - one question first thought:

When you say `sell order` here do you mean the listing of the item for sale in the marketplace or the execution of the order by the buyer?

Reply:
the listing of the item for sale in the marketplace (sellOrder == assetForSale)

Reply:
But we're in the `My assets` screen for this comment thread. So the tx hash of the listing of the item for sale in the marketplace has nothing to do with the tx hash we'll want to use to build the Etherscan link on this screen. Does that mean there may be a bug for this feature?

Reply:
Oh, good question. I've misunderstood the terms here. I've used `sellOrders` wrongly.
Actually, assets from an address are being got using events too. We are using `Transfer` events to get that, so, the txhash applies here too.

Reply:
Got it - that makes sense. Thanks! Resolving this.